app/src/main/java/com/github/houseorganizer/houseorganizer/panels/MainScreenActivity.java

```diff
@@ -70,8 +85,16 @@ protected void onCreate(Bundle savedInstanceState) {
 
         mUser = FirebaseAuth.getInstance().getCurrentUser();
         db = FirebaseFirestore.getInstance();
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
+
+        loadHouse = getIntent().hasExtra("LoadHouse");
+
+        if(!loadHouse) loadData();
+        if(loadHouse && LocationHelpers.checkLocationPermission(this, this)){ // TODO find a way for not having 2 "this"
```

Comment:
you might be able to call Activity.getContext or something similar, that way you only need one "this"

Reply:
Yes done thanks !